src/main/java/org/testng/xml/XmlWeaver.java

```diff
@@ -76,6 +76,13 @@ private static IWeaveXml attemptDefaultImplementationInstantiation() {
         if (clazz.equals(CommentDisabledXmlWeaver.class.getName())) {
             return new CommentDisabledXmlWeaver();
         }
-        return null;
+        if (testMode) {
+            return null;
+        }
+        Class<?> clazzName = ClassHelper.forName(clazz);
```

Comment:
Could you explain this change?

Reply:
For test mode we aren't passing any implementation class names as Fully qualified class names. That is why its ok to return a `null` value. But in production mode, a user could also try and pass in an implementation which we can use.

Reply:
Ok :+1: